File: PS/Programmers/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    answer = 0
    que = []
    max = 0
    for i in range(len(truck_weights)):
        size = truck_weights[i];
        while True:
            if len(que) == 0:
                que.append(size)
                max+=size
                answer+=1
                break
            elif len(que) == bridge_length:
                max -= que.pop(0)
            else:
                if (size+max)>weight:
                    que.append(0)
                    answer+=1
                else:
                    que.append(size)
                    max+=size
                    answer+=1
                    break
    return answer+bridge_length

File: PS/Programmers/다리를 지나는 트럭.cc
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <queue>
#include<algorithm>
using namespace::std;

int solution(int bridge_length, int weight, vector<int> truck_weights) {
    int answer = 0;
    queue<int> q;
    int max = 0, size = 0;
    for (int i = 0; i < truck_weights.size(); i++)
    {
        size = truck_weights[i];
        while (1)
        {
            if (q.empty())
            {
                q.push(size); max += size; answer++;
                break;
            }
            else if (q.size() == bridge_length)
            {
                max -= q.front();
                q.pop();
            }
            else
            {
                if (size + max > weight)
                {
                    q.push(0); answer++;
                }
                else
                {
                    q.push(size); max += size;
                    answer++;
                    break;
                }

            }
        }
    }
   
    return answer+bridge_length;
}